csrc/quantization/cutlass_w8a8/scaled_mm_c2x.cu

```diff
@@ -57,6 +57,29 @@ void cutlass_scaled_mm_sm75(torch::Tensor& out, torch::Tensor const& a,
   }
 }
 
+void cutlass_scaled_mm_azp_sm75(torch::Tensor& out, torch::Tensor const& a,
+                                torch::Tensor const& b,
+                                torch::Tensor const& a_scales,
+                                torch::Tensor const& b_scales,
+                                torch::Tensor const& azp_adj,
+                                c10::optional<torch::Tensor> const& azp,
+                                c10::optional<torch::Tensor> const& bias) {
+  TORCH_CHECK(a_scales.dtype() == torch::kFloat32);
+  TORCH_CHECK(b_scales.dtype() == torch::kFloat32);
+  TORCH_CHECK(!bias || bias->dtype() == out.dtype(),
+              "currently bias dtype must match output dtype ", out.dtype());
+  TORCH_CHECK(azp_adj.dtype() == torch::kInt32);
+
+  if (azp) {
+    TORCH_CHECK(azp->dtype() == torch::kInt32);
```

Comment:
It looks like these TORCH_CHECKs always the same. If so, could you move them into `cutlass_scaled_mm_azp` instead of having them be duplicated in each arch-specific function?

Reply:
Should I do that for scales as well?

Reply:
I'll do it in a cleanup PR later

Reply:
Yeah, scales as well would be good to factor out.